Artifacts/windows-visualstudio/artifact.ps1

```diff
@@ -0,0 +1,122 @@
+Param(
+    [Parameter(Mandatory=$true)]
+    [ValidateSet("2015","2017")] 
+    [string] $version,
+
+    [Parameter(Mandatory=$true)]
+    [ValidateSet("Professional","Enterprise")] 
+    [string] $sku,
+
+    [Parameter()]
+    [string] $installerArgs
+)
+
+function DownloadToFilePath ($downloadUrl, $targetFile)
+{
+    Write-Output ("Downloading installation files from URL: $downloadUrl to $targetFile")
+    $targetFolder = Split-Path $targetFile
+
+    if((Test-Path -path $targetFile))
+    {
+        Write-Output "Deleting old target file $targetFile"
+        Remove-Item $targetFile -Force | Out-Null
+    }
+
+    if((Test-Path -path $targetFolder) -eq $false)
+    {
+        Write-Output "Creating folder $targetFolder"
+        New-Item -ItemType Directory -Force -Path $targetFolder | Out-Null
+    }
+
+    #Download the file
+    $downloadAttempts = 0
+    do
+    {
+        $downloadAttempts++
+
+        try
+        {
+            $WebClient = New-Object System.Net.WebClient
+            $WebClient.DownloadFile($downloadUrl,$targetFile)
+            break
+        }
+        catch [Exception]
+        {
+            Write-Output "Caught exception during download..."
+            if ($_.Exception.InnerException){
+                $exceptionMessage = $_.InnerException.Message
+                Write-Output "InnerException: $exceptionMessage"
+            }
+            else {
+                $exceptionMessage = $_.Message
+                Write-Output "Exception: $exceptionMessage"
+            }
+        }
+
+    } while ($downloadAttempts -lt 5)
+
+    if($downloadAttempts -eq 5)
+    {
+        Write-Error "Download of $downloadUrl failed repeatedly. Giving up."
+    }
+}
+
+Write-Output "Installing Visual Studio $version $sku"
+$logFolder = Join-path -path $env:ProgramData -childPath "DTLArt_VS"
+
+if($version -eq '2015' )
+{
+    $vsLog = Join-Path $logFolder "VSInstall.log"
+    $argumentList = "$installerArgs /Quiet /NoRestart /Log $vsLog"
+
+    if($sku -eq 'Professional') {
+        $downloadUrl = 'http://go.microsoft.com/fwlink/?LinkId=615435'
+    }
+    elseif($sku -eq 'Enterprise') {
+        $downloadUrl = 'http://go.microsoft.com/fwlink/?LinkId=615437'
+    }
+}
+elseif ($version -eq '2017')
+{
+    if($sku -eq 'Professional') {
+        $argumentList = "--quiet --norestart --wait --add Microsoft.VisualStudio.Workload.Azure --add Microsoft.VisualStudio.Workload.Data --add Microsoft.VisualStudio.Workload.ManagedDesktop --add Microsoft.VisualStudio.Workload.NativeDesktop --add Microsoft.VisualStudio.Workload.NetCoreTools --add Microsoft.VisualStudio.Workload.NetWeb --includeRecommended"
```

Comment:
Not that it will be a problem now, but if you want to make sure the `$argumentList` doesn't get expanded, you may want to surround it by single quotes instead of double quotes.

Reply:
Sure. I can use single quotes here